nomenclature/processor/region.py

```diff
@@ -667,3 +671,17 @@ def _check_exclude_region_overlap(values: Dict, region_type: str) -> Dict:
             region=overlap, region_type=region_type, file=values["file"]
         )
     return values
+
+
+class ReverseRegionProcessor(RegionProcessor):
+    def apply(self, df: pyam.IamDataFrame) -> pyam.IamDataFrame:
+        model_dfs = []
+        for model in df.model:
+            model_df = df.filter(model=model)
+            if mapping := self.mappings.get(model):
+                # remove common regions
+                model_df = model_df.filter(
+                    region=mapping.common_region_names, keep=False
+                ).rename(region=mapping.reverse_rename_mapping)
```

Comment:
Maybe better readability?
```suggestion
                model_df = (
                    model_df
                    .filter(region=mapping.common_region_names, keep=False)
                    .rename(region=mapping.reverse_rename_mapping)
                )
```